eclair-core/src/main/scala/fr/acinq/eclair/blockchain/bitcoind/rpc/BitcoinCoreClient.scala

```diff
@@ -224,12 +225,12 @@ class BitcoinCoreClient(val rpcClient: BitcoinJsonRPCClient, val onChainKeyManag
       val JInt(confirmations) = tx \ "confirmations"
       // while transactions are still in the mempool, block hash will no be included
       val blockHash = tx \ "blockhash" match {
-        case JString(blockHash) => ByteVector32.fromValidHex(blockHash)
-        case _ => ByteVector32.Zeroes
+        case JString(blockHash) => BlockHash(ByteVector32.fromValidHex(blockHash))
```

Comment:
Good catch! It's impossible to test on regtest, since we can't really distinguish between a hash and an id since there is no difficulty :sweat_smile: 

Fixed in https://github.com/ACINQ/eclair/pull/2742/commits/23d44b150c8fa27ef5539dabf5688fb6aeb13aff